jenkins-cicd-vmss/azuredeploy.json

```diff
@@ -0,0 +1,173 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "jenkinsVMAdminUsername": {
+      "type": "string",
+      "defaultValue": "jenkinsadmin",
+      "minLength": 1,
+      "metadata": {
+        "description": "User name for the Jenkins Virtual Machine."
+      }
+    },
+    "jenkinsVMAdminPassword": {
+      "type": "securestring",
+      "minLength": 6,
+      "metadata": {
+        "description": "Password for the Jenkins Virtual Machine."
+      }
+    },
+    "jenkinsDnsPrefix": {
+      "type": "string",
+      "minLength": 3,
+      "metadata": {
+        "description": "Unique DNS Name for the Public IP used to access the Jenkins Virtual Machine."
+      }
+    },
+    "jenkinsReleaseType": {
+      "type": "string",
+      "allowedValues": [
+        "LTS",
+        "weekly",
+        "verified"
+      ],
+      "defaultValue": "LTS",
+      "metadata": {
+        "description": "The Jenkins release type."
+      }
+    },
+    "repositoryUrl": {
+      "type": "string",
+      "defaultValue": "https://github.com/Azure/azure-quickstart-templates",
+      "minLength": 1,
+      "metadata": {
+        "description": "GitHub repository URL, e.g. https://github.com/xxx/yyy."
+      }
+    },
+    "clientId": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "Client id for Azure service principal."
+      }
+    },
+    "clientSecret": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "Client secret for Azure service principal."
+      }
+    },
+    "VMResourceGroup": {
```

Comment:
Our requirements and implementation require this to be a separate resource group, so can we keep it as it is?

Reply:
How is the rg used?  it doesn't appear to be enforced in anyway...  users could easily specify the current resourceGroup

Reply:
We're building a CI/CD process via Jenkins job, and the VMSS resources will be deleted and then recreated by Terraform template (other than ARM template) for each build automatically, so we decided to manage theses resources in a separate resource group, or does that work to you if we remove that parameter and use such name for the VMSS rg [concat(resourceGroup().name, '-VMSS')]?

Reply:
Just removed this parameter and use [concat(resourceGroup().name, '-VMSS')] for that resource group alternatively, to avoid the user use the same RG name as Jenkins, which would cause troubles

Reply:
What if that resourceGroup already exists?

Reply:
The existing resource group will be deleted when new build is triggered, this is following the business requirements for CI/CD, the name generated by the rules above should not exist as usual, but I just updated it to append an unique string to make that's more reliable.

[concat(resourceGroup().name, '_VMSS_', uniqueString(resourceGroup().id))]